### DISCUSSION TITLE
(Vive Tracker 3.0) => Having the ability to rotate 90°

### DISCUSSION THREAD

**Discussion by Chpouky** [opentrack/opentrack#1489] [Ideas]
Hey everyone !

I'm trying to use my Vive Tracker 3.0 for headtracking, but unfortunately the orientation of the 3.0 version makes it so that it needs to be rotated 90degrees forward for the different axis to be correct (if used layed flat on your head, the axis are wrong and Yaw is doing a Roll instead). The option to offset orientation for the game to interpret is not fixing this.

I tried messing around with the transform settings but couldn't make it work.

**Comment by sthalik** [comment]
See if #1483 fixed this issue.

**Reply by Chpouky** [reply]
Thanks for the quick response ! Unfortunately it's not working :/

**Reply by sthalik** [reply]
Do I understand correctly that in this new version tracking doesn't work at all?

**Reply by Chpouky** [reply]
Tracking works perfectly for me with the latest opentrack build (as well as the one you sent me).

It's just the orientation that's wrong, but this is not related to an OpenTrack bug, because the tracker 3.0 has weird axis orientation.
It needs to be held like this for the axis to be correct (middle image):

![image](https://user-images.githubusercontent.com/72340012/172049947-e6e685aa-0653-4248-bdc0-87f9093b3d1e.png)

But I want to change the axis in opentrack so that I can lay my tracker flat on my headphones.

**Reply by sthalik** [reply]
What happens if you use 90 degrees yaw as a custom center pose and enable output remap to fix swapped yaw and roll? Also switch centering mode to VR from roll compensated, which is the default.

**Reply by Chpouky** [reply]
This is not fixing it unfortunately :/ I didn't know about the centering mode, but that didn't fix it.

**Reply by Chpouky** [reply]
Like, swapping Yaw and Roll is not making a difference, and when I recenter, the squid is not oriented properly facing me, it's looking up.

**Reply by sthalik** [reply]
The octopus not being centered when you explicitly recenter -- is this the case in the official release build as well?

**Reply by Chpouky** [reply]
I FIXED IT !

The mode you have to use for centering is "Point" instead of anything else ! No need to change axis or add a transform.

**Reply by sthalik** [reply]
There's something else you could try out. SteamVR centering is currently only enabled for HMDs. Maybe changing this could help solve this without the need for you to make any configuration changes? New build: https://www.dropbox.com/s/jjpmrzagjfoqiot/opentrack-test-20220605_01.7z

**Reply by Chpouky** [reply]
No need ^^ Changing centering mode to "Point" fixed my orientation issue. Now it's working perfectly as intended !
